### PR TITLE
fix(CI): Force clippy to run on the stable channel

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This just ensures that we run clippy on the stable channel, so we don't get inconsistent reporting from nightly-only lints.

### CODE REVIEW

**Review by miker83z — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

**Review by lzpap — CHANGES_REQUESTED** [review]
What we uncovered yesterday is that https://github.com/iotaledger/iota/pull/917 introduced new lint headers that do not work well with some clippy versions on some runners. Take a look at [this run for example](https://github.com/iotaledger/iota/actions/runs/9942349320/job/27463701122):
 - clippy is added via rustup, and the job says: `info: component 'clippy' for target 'x86_64-unknown-linux-gnu' is up to date`
 - clippy doesn't know the `#[allow(clippy::const_is_empty)]` lint and the job fails

The `unknown` lints were added in the latest clippy release ([release notes](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#new-lints)), so it seems that some runners don't have the newest clippy version.
 - either we remove these new lints from the code, or
 - make sure that every runner has updated clippy. @shufps could you help with that? 
 
Until the issue is resolved we should not merge this PR.

**Review by lzpap — COMMENTED** [review]
> We should remove them, but that makes me confused about why the CI passed for this PR.

Because you got a runner that has the updated one. [This one](https://github.com/iotaledger/iota/actions/runs/10058091033/job/27800610587?pr=1172) got a runner that was not.

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by lzpap — APPROVED** [review]

**Pull request merged**
